features/telemetry/protocols/collection/src/main/java/org/opennms/netmgt/telemetry/protocols/collection/AbstractPersistingAdapter.java

```diff
@@ -188,10 +203,46 @@ public void handleMessageLog(TelemetryMessageLog messageLog) {
                 LOG.trace("Persisting collection set: {} for message: {}", collectionSet, message);
                 final Persister persister = persisterFactory.createPersister(EMPTY_SERVICE_PARAMETERS, repository);
                 collectionSet.visit(persister);
+
+                // Thresholding
+                try {
+                    if (isThresholdingEnabled) {
+                        ThresholdingSession session = getSessionForAgent(result.getAgent(), repository);
+                        session.accept(collectionSet);
+                    }
+                } catch (ThresholdInitializationException e) {
+                    LOG.warn("Failed Thresholding of CollectionSet: {} ", e.getMessage());
```

Comment:
Can we include some of the agent details in the log message when this happens? i.e. What agent was affected?